pkg/executor/api.go

```diff
@@ -249,6 +249,62 @@ func (executor *Executor) unTapService(w http.ResponseWriter, r *http.Request) {
 	w.WriteHeader(http.StatusOK)
 }
 
+// isValidHandler is very similar to getServiceForFunctionAPI, with the difference that the response returned doens't contain
+// the address, but instead returns "true" or "false", depending if the function service is valid or not.
+func (executor *Executor) isValidHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	logger := otelUtils.LoggerWithTraceID(ctx, executor.logger)
+
+	body, err := io.ReadAll(r.Body)
+	if err != nil {
+		http.Error(w, "Failed to read request", http.StatusInternalServerError)
+		return
+	}
+
+	// get function metadata
+	fn := &functionWithAddress{}
+	err = json.Unmarshal(body, &fn)
+	if err != nil {
+		logger.Error("Error unmarshelling: ", zap.Any("error", err))
+		http.Error(w, "Failed to parse request", http.StatusBadRequest)
+		return
+	}
+
+	t := fn.Function.Spec.InvokeStrategy.ExecutionStrategy.ExecutorType
+	et := executor.executorTypes[t]
+	funct := fn.Function
+
+	if t == fv1.ExecutorTypePoolmgr && !funct.Spec.OnceOnly {
+		fsvc, err := et.GetFuncSvcFromPoolCacheByAddress(ctx, &fn.Function, fn.SvcAddress)
+
+		// check if its a cache hit (check if there is already specialized function pod that can serve another request)
+		if err == nil {
+			if et.IsValid(ctx, fsvc) {
+				// return true
+				logger.Debug("served from cache", zap.String("name", fsvc.Name), zap.String("address", fsvc.Address))
```

Comment:
This message isn't very helpful, perhaps: "found matching function service in cache"

Reply:
I agree. I've went with your suggestion. Please take a look